File: mobile/src/hiddenApiStubs/java/android/net/wifi/OuiKeyedData.java
package android.net.wifi;

import android.os.Parcelable;
import android.os.PersistableBundle;
import androidx.annotation.RequiresApi;

@RequiresApi(35)
public abstract class OuiKeyedData implements Parcelable {
    public abstract PersistableBundle getData();

    public abstract int getOui();

    public static final class Builder {
        public Builder(int oui, PersistableBundle data) {
        }

        public OuiKeyedData build() {
            throw new UnsupportedOperationException();
        }
    }
}

File: mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/OuiKeyedData.kt
package be.mygod.vpnhotspot.net.wifi

import android.os.Parcelable
import android.os.PersistableBundle
import androidx.annotation.RequiresApi
import kotlinx.parcelize.Parcelize

@Parcelize
@JvmInline
@RequiresApi(35)
value class OuiKeyedData(val inner: Parcelable) : Parcelable {
    companion object {
        private val clazz by lazy { Class.forName("android.net.wifi.OuiKeyedData") }
        private val getData by lazy { clazz.getDeclaredMethod("getData") }
        private val getOui by lazy { clazz.getDeclaredMethod("getOui") }
        private val builderClass by lazy { Class.forName("android.net.wifi.OuiKeyedData\$Builder") }
        private val newBuilder by lazy {
            builderClass.getConstructor(Int::class.java, PersistableBundle::class.java)
        }
        private val build by lazy { builderClass.getDeclaredMethod("build") }

        operator fun invoke(oui: Int, data: PersistableBundle) =
            OuiKeyedData(build(newBuilder.newInstance(oui, data)) as Parcelable)
    }

    val data get() = getData(inner) as PersistableBundle
    val oui get() = getOui(inner) as Int
}
